Doc/howto/functional.rst

```diff
@@ -529,18 +517,15 @@ Passing values into a generator
 
 In Python 2.4 and earlier, generators only produced output.  Once a generator's
 code was invoked to create an iterator, there was no way to pass any new
-information into the function when its execution is resumed.  You could hack
-together this ability by making the generator look at a global variable or by
-passing in some mutable object that callers then modify, but these approaches
-are messy.
+information into the function when its execution is resumed.
 
-In Python 2.5 there's a simple way to pass values into a generator.
-:keyword:`yield` became an expression, returning a value that can be assigned to
-a variable or otherwise operated on::
+In Python 2.5, :keyword:`yield` became an expression, returning a value that
+can be assigned to a variable or otherwise operated on, providing a simple
+way to pass values into a generator::
 
     val = (yield i)
 
-I recommend that you **always** put parentheses around a ``yield`` expression
+It's recommended that you **always** put parentheses around a ``yield`` expression
```

Comment:
The current wording is more concise. It's also more precise -- with your wording, I'm left wondering who exactly is recommending this? Lots of things in this article are just one person's opinion, and never claim to be anything else, but with your rewording here, it makes it sound like it's the official pronouncement from on high about the proper programming style for Python. Removing the usage of the first person throughout the article really muddies the water here about the purpose of this article, in my opinion.

Reply:
If they are just one person's opinion, they don't belong in the documentation. The documentation's contract with the user is exactly that: that it is the authoritative pronouncement from on-high about programming in Python. We would never allow the code to be so permissive and lax, to represent merely the way someone feels it's good to do it, and we should hold documentation to the same standards.

Reply:
> If they are just one person's opinion, they don't belong in the documentation.

Says who? Again, that's _not_ how the CPython project has traditionally thought of HOWTO documents, and it's _not_ what the `Doc/howto/` directory in CPython has historically been used for. If you want to substantially rewrite a large portion of the articles in `Doc/howto/` based on a new, much stricter idea of what it means for a document to be a HOWTO guide, then I honestly think you should try to achieve a broader consensus among the community before going ahead with something like this.  I haven't seen any discussions on Discourse about this, for example.

Personally, I think it's great that we have a space in the CPython documentation for more subjective, personal reflections on programming practices. I'd much prefer we say "it's not a how-to" and move it to some other directory than try to turn the document into something it's not.

Reply:
> > If they are just one person's opinion, they don't belong in the documentation.
> 
> Says who? Again, that's _not_ how the CPython project has traditionally thought of HOWTO documents, and it's _not_ what the `Doc/howto/` directory in CPython has historically been used for. If you want to substantially rewrite a large portion of the articles in `Doc/howto/` based on a new, much stricter idea of what it means for a document to be a HOWTO guide, then I honestly think you should try to achieve a broader consensus among the community before going ahead with something like this. I haven't seen any discussions on Discourse about this, for example.

I understood - I believe - that there _is_ consensus on adopting Diátaxis for Python's documentation. There have been discussions, I ran a couple of workshops with the documentation team, and so on. 

It's a big project, and it's going to take a long time. My goal is to help set new standards of quality in Python's documentation. It is going to change a lot, and set new expectations. There isn't any other way around this.

> Personally, I think it's great that we have a space in the CPython documentation for more subjective, personal reflections on programming practices. I'd much prefer we say "it's not a how-to" and move it to some other directory than try to turn the document into something it's not.

Yes, that is precisely what I described as the goal in the original commit for this pull request. These are steps along the way. But even explanation-type documentation, which is what this particular document is, needs to have authority. Otherwise it is simply not documentation.

Reply:
> I understood - I believe - that there is consensus on adopting Diátaxis for Python's documentation.

I've liked a lot of what I've seen from the Diataxis work on the CPython docs -- I think it's resulted in some very worthwhile improvements to the docs, and I'm excited to see it continue! We need to consider each change individually, however. There may be broad consensus about using Diataxis as a framework for the CPython docs overall, but I don't really feel like I've seen any kind of consensus-building about these older HOWTO documents.

> I ran a couple of workshops with the documentation team

(Super minor nitpick, but there is no "documentation team". There's a docs _community_ -- a group of people who are interested and enthusiastic about improving the CPython docs. But docs contributions come from a wide variety of people, many of whom aren't part of that community, and there are lots of core developers interested in documentation who aren't part of that community. I think it would be a mistake to assume that the people in that community are _necessarily_ representative of the Python community as a whole 🙂)

Reply:
> Yes, that is precisely what I described as the goal in the original commit for this pull request.

I don't understand -- if the goal is to move this document out of the `Doc/howto/` directory, then why do you keep on making arguments based on what a HOWTO is "supposed to be"? If the hope is to reclassify this document altogether, then surely we should do that first, so that we have a clear idea of what the document is meant to be while we're editing it?